DEV mode doesn't work when path contains international characters, regression against 1.6.1.Final / 1.3.4.Final

DEV mode doesn't work when path contains international characters, regression against 1.6.1.Final

When the path contains `ěščřžýáíéůú`or `Îñţérñåţîöñåļîžåţîờñ` I receive error `Prepare phase of the quarkus-maven-plugin threw an error: Failed to load current project ...`

This is regression comparing to Quarkus 1.6.1.Final which works fine

**Part of stacktrace**
```
Caused by: io.quarkus.bootstrap.resolver.maven.BootstrapMavenException: Failed to locate project /Users/rsvoboda/ěščřžýáíéůú/getting-started-1.7/pom.xml in the loaded workspace
    at io.quarkus.bootstrap.resolver.maven.workspace.LocalProject$WorkspaceLoader.load (LocalProject.java:127)
    at io.quarkus.bootstrap.resolver.maven.workspace.LocalProject.loadWorkspace (LocalProject.java:202)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.resolveCurrentProject (BootstrapMavenContext.java:251)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.<init> (BootstrapMavenContext.java:144)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init> (MavenArtifactResolver.java:103)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init> (MavenArtifactResolver.java:52)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver$Builder.build (MavenArtifactResolver.java:88)
    at io.quarkus.maven.CodeGenMojo.doExecute (CodeGenMojo.java:91)
    at io.quarkus.maven.CodeGenMojo.execute (CodeGenMojo.java:62)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:119)
    at io.quarkus.maven.DevMojo.triggerPrepare (DevMojo.java:413)
    at io.quarkus.maven.DevMojo.handleAutoCompile (DevMojo.java:405)
    at io.quarkus.maven.DevMojo.execute (DevMojo.java:273)
```

**Reproducer:**
Prepare directory
```
mkdir ěščřžýáíéůú && cd ěščřžýáíéůú
## or 
mkdir Îñţérñåţîöñåļîžåţîờñ && cd Îñţérñåţîöñåļîžåţîờñ
```

Generated and run the app
```
## 1.7 - FAIL
mvn io.quarkus:quarkus-maven-plugin:1.7.0.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started-1.7 \
    -DclassName="org.acme.getting.started.GreetingResource" \
    -Dpath="/hello"
cd getting-started-1.7
mvn quarkus:dev

## 1.6 - OK
mvn io.quarkus:quarkus-maven-plugin:1.7.0.Final:create \
    -DplatformVersion=1.6.0.Final \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started-1.6 \
    -DclassName="org.acme.getting.started.GreetingResource" \
    -Dpath="/hello"
cd getting-started-1.6
mvn quarkus:dev
```

--- a/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
+++ b/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
@@ -137,8 +137,13 @@ public class LocalProject {
             if (parent != null) {
                 parent.modules.add(project);
             }
-            if (workspace.getCurrentProject() == null && currentProjectPom.getParent().equals(project.getDir())) {
-                workspace.setCurrentProject(project);
+            try {
+                if (workspace.getCurrentProject() == null
+                        && Files.isSameFile(currentProjectPom.getParent(), project.getDir())) {
+                    workspace.setCurrentProject(project);
+                }
+            } catch (IOException e) {
+                throw new BootstrapMavenException("Failed to load current project", e);
             }
             final List<String> modules = project.getRawModel().getModules();
             if (!modules.isEmpty()) {

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/JarRunnerIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/JarRunnerIT.java
@@ -46,6 +46,43 @@ import io.quarkus.utilities.JavaBinFinder;
 @DisableForNative
 public class JarRunnerIT extends MojoTestBase {
 
+    /**
+     * Tests that a Quarkus project builds fine if the project is hosted in a directory
+     * path that contains non-ASCII characters
+     *
+     * @throws MavenInvocationException
+     * @throws IOException
+     * @see <a href="https://github.com/quarkusio/quarkus/issues/11511"/>
+     */
+    @Test
+    public void testNonAsciiDir() throws Exception {
+        final File testDir = initProject("projects/classic", "projects/ěščřžýáíéůú");
+        final RunningInvoker running = new RunningInvoker(testDir, false);
+
+        final MavenProcessInvocationResult result = running.execute(Arrays.asList("install", "-DskipTests"),
+                Collections.emptyMap());
+        await().atMost(1, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
+        assertThat(running.log()).containsIgnoringCase("BUILD SUCCESS");
+        running.stop();
+
+        File output = new File(testDir, "target/output.log");
+        output.createNewFile();
+
+        Process process = doLaunch(new File(testDir, "target"), Paths.get("acme-1.0-SNAPSHOT-runner.jar"), output,
+                Collections.emptyList()).start();
+        try {
+            // Wait until server up
+            dumpFileContentOnFailure(() -> {
+                await().pollDelay(1, TimeUnit.SECONDS)
+                        .atMost(1, TimeUnit.MINUTES).until(() -> DevModeTestUtils.getHttpResponse("/app/hello/package", 200));
+                return null;
+            }, output, ConditionTimeoutException.class);
+        } finally {
+            process.destroy();
+        }
+
+    }
+
     @Test
     public void testThatJarRunnerConsoleOutputWorksCorrectly() throws MavenInvocationException, IOException {
         File testDir = initProject("projects/classic", "projects/project-classic-console-output");
@@ -268,10 +305,15 @@ public class JarRunnerIT extends MojoTestBase {
     }
 
     private ProcessBuilder doLaunch(Path jar, File output) throws IOException {
-        return doLaunch(jar, output, Collections.emptyList());
+        return doLaunch(null, jar, output, Collections.emptyList());
     }
 
     private ProcessBuilder doLaunch(Path jar, File output, Collection<String> vmArgs) throws IOException {
+        return doLaunch(null, jar, output, vmArgs);
+    }
+
+    private ProcessBuilder doLaunch(final File workingDir, final Path jar, File output, Collection<String> vmArgs)
+            throws IOException {
         List<String> commands = new ArrayList<>();
         commands.add(JavaBinFinder.findBin());
         commands.addAll(vmArgs);
@@ -280,6 +322,9 @@ public class JarRunnerIT extends MojoTestBase {
         // write out the command used to launch the process, into the log file
         Files.write(output.toPath(), commands);
         ProcessBuilder processBuilder = new ProcessBuilder(commands.toArray(new String[0]));
+        if (workingDir != null) {
+            processBuilder.directory(workingDir);
+        }
         processBuilder.redirectOutput(ProcessBuilder.Redirect.appendTo(output));
         processBuilder.redirectError(ProcessBuilder.Redirect.appendTo(output));
         return processBuilder;

--- a/integration-tests/maven/src/test/resources/projects/classic/pom.xml
+++ b/integration-tests/maven/src/test/resources/projects/classic/pom.xml
@@ -65,6 +65,8 @@
         <executions>
           <execution>
             <goals>
+              <goal>prepare</goal>
+              <goal>prepare-tests</goal>
               <goal>build</goal>
             </goals>
           </execution>

